src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -1426,23 +1426,51 @@ class LedgerMasterImp
         if (!referenceLedger || (referenceLedger->getLedgerSeq() < index))
             return ledgerHash; // Nothing we can do. No validated ledger.
 
-        // See if the hash for the ledger we need is in the reference ledger
+        // Is the hash for the ledger we need in the reference ledger?
         ledgerHash = referenceLedger->getLedgerHash (index);
-        if (ledgerHash.isZero())
+
+        // Yes
+        if (ledgerHash.isNonZero())
+            return ledgerHash;
+
+        // No. We need to get another ledger that might have it. That ledger
+        // must meet two requirements:
+        //  1) It must have the hash of the ledger we are looking for.
```

Comment:
You're using this idea of "the ledger we're looking for" all the way through this explanation - which keeps dragging "us" into it.  If you gave it a name, then I think "we" would vanish.  :-)

Why not call it the "requested ledger"?  Also, "have" is ambiguous... what about "contain"?

    // 1) It must contain the hash of the requested ledger.